### ISSUE TITLE
Why do I get “RuntimeWarning: numpy.dtype size changed” warning when upgrading MLflow?

### ISSUE THREAD

**Issue by emres** [mlflow/mlflow#274]
I've just upgraded mlflow from **0.2.1** to **0.4.2**, and now I'm getting the following warning:

```
/usr/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
return f(*args, **kwds)
```

mlflow runs, but I wonder why this warning suddenly started to appear now. Seems like it might be related to [this](https://stackoverflow.com/a/40846742/236007) and [this](https://github.com/ContinuumIO/anaconda-issues/issues/6678), but I wanted to check with mlflow team if there's a way to get rid of this warning.

**Comment by mateiz** [comment]
It might have something to do with how your Python environment is set up. How did you set up MLflow? Maybe blowing away the environment to create a new one would solve it, but it would be great to hear how it got to this point. I've definitely seen this issue too sometimes.

**Comment by emres** [comment]
Well, what I did to install version 0.2.1 was to create a Python 3 virtual environment using

 `python3 -m venv mlflow`

and then activate the environment, and then 

`pip install mlflow`

And it worked fine without any warnings, but started to give those warnings yesterday, after I did

`pip install --upgrade mlflow`

I also have conda 4.5.9 installed on my system, I don't know if this might be related.

Based on your suggestion, I'll try to create a new virtual environment and install MLflow there to see if I get the same warning.

**Comment by emres** [comment]
I've created a new virtual Python3 environment and installed mlflow there, only to receive the very same warning:

```
python3 -m venv mlflow-emre
source /home/emre/mlflow-emre/bin/activate
pip install mlflow
mlflow --version
/opt/anaconda/anaconda3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
mlflow, version 0.4.2
```

**Comment by mateiz** [comment]
OK, got it. This seems to be from having libraries that were compiled against different version of NumPy for some reason, but it's not clear why it would happen. It could also be due to a PYTHONPATH that somehow includes libraries from multiple installations of Python. Anyway we'll post here if we find the cause, because we have seen this too.

**Comment by aarondav** [comment]
If this is relevant:
https://github.com/ContinuumIO/anaconda-issues/issues/6678#issuecomment-337279157

then the warning may be intentional/unavoidable from the conda side. We could try suppressing these warnings in mlflow...

**Comment by emres** [comment]
I think that's the case. I'm going to close this issue, because this warning isn't stopping me in any way to work with MLflow.

**Comment by aarondav** [comment]
Btw, we fixed this in #317.
